ddtrace/profiling/collector/memalloc.py

```diff
@@ -148,7 +148,13 @@ def snapshot(self):
         )
 
     def collect(self):
-        events, count, alloc_count = _memalloc.iter_events()
+        try:
+            events, count, alloc_count = _memalloc.iter_events()
```

Comment:
is this a related change?

Reply:
We could take this out into its own PR. This prevents a race between the periodic memory collector service and the `_memalloc.stop` clean-up on shutdown from being noisy. This is not related to just gevent and might actually fix #3121.

Reply:
https://github.com/DataDog/dd-trace-py/pull/5182